x.consolidation/consolidation.md

```diff
@@ -0,0 +1,517 @@
+# UTXO Consolidation
+
+Unspent Transaction Output (UTXO) consolidation is the act of taking the
+bitcoin value stored in several UTXOs and *consolidating* it into one
+UTXO.  This lowers the number of UTXOs you control while keeping your
+balance roughly the same, increasing the chance that you'll be able to
+create smaller and more affordable transactions at a later time when
+you need to pay a higher feerate in order to obtain a fast confirmation.
+
+## A consolidation example
+
+Imagine a transaction as a rectangle whose height is its feerate, whose
+length is its size (block weight), and whose area is its total fee
+(feerate times size).  For example, illustrated below is a simple
+transaction that spends from one P2WPKH UTXO (input) to two P2WPKH
+outputs---one output to a client and one as change back to the spender.
+
+![Example of a 1-input, 2-output transaction](img/consolidation-example-1in2out.png)
+
+Most wallets eventually encounter a case where they don't have enough
+value in a single UTXO to afford the amount they want to spend.  This
+requires that they add more value to the transaction by spending
+additional UTXOs.  For example, let's spend two extra P2WPKH UTXOs in
+our example transaction:
+
+![Example of a 3-input, 2-output transaction](img/consolidation-example-3in2out.png)
+
+Although that seemed like a minor change, it almost doubled the size and
+cost of our transaction.  This is where consolidation can be useful.  By
+preparing ahead, we can can *consolidate* the three UTXOs of our large
+transaction into a single UTXO using a low-feerate transaction spent from
+ourself to ourself.  Then we can pay the client using a single UTXO in a
+high-feerate transaction that has an identical shape to the first
+example transaction above:
+
+![Example of a 3-input, 1-output transaction followed by a 1-input, 2-output transaction](img/consolidation-example-3in1out-1in2out.png)
+
+Comparing the total costs, we find that the two transaction version is
+40% cheaper than the single high-feerate transaction with three
+UTXOs.  This is possible even though the two-transaction version uses
+more total block space.
+
+If the consolidation transaction confirms before you send the second
```

Comment:
> it seems somewhat implied that the consolidating tx would be confirmed

Yeah, it was implied earlier, but here I make that assumption explicit for accuracy.  It's also important because, later in the chapter, we describe fee bumping strategies to deal with consolidations that haven't confirmed yet.  I'll see if I can rephrase this.

Reply:
Rewording looks good.